### PR TITLE
Replace pip with Poetry and upgrades for Python 3.10

### PR DESCRIPTION
**Pull request by markspolakovs**
Basically what I needed to do to get the Python bits running on 3.10, as well as getting frustrated with dealing with `requirements.txt`.

The changes should be fairly safe, just formatting (vscode ran black automatically... we should probably run it as part of CI though) and a few import changes. Does mean we'll need to set up a new venv on dolby using poetry, but this is fine.

Only risky thing is that we may need to upgrade dolby's python.

### CODE REVIEW

**Comment by michael-grace** [comment]
whack it in docker, it doesn't rely on dolby [i think]

**Comment by michael-grace** [comment]
wait, wrong repo ignore that

**Comment by mstratford** [comment]
Note I don't think black is quite PEP8 compliant, but it's certainly better than none.

File: README.md [hunk 1]
```diff
@@ -10,24 +10,22 @@ The clientside is written in TypeScript using React and Redux, the serverside is
 
 Client:
 
- * Node.js and Yarn 1.x
+- Node.js and Yarn 1.x
 
 Server:
 
-* Python 3.7-3.9 (note: Python 3.10 is not supported)
+- Python 3.7-3.9 (note: Python 3.10 is not supported)
```
Comment (mstratford) [inline README.md:17]: I guess this is now 3.7-3.10?

**Comment by markspolakovs** [comment]
It claim's in the README that it's PEP8 compliant at least (and I think it's a bit stricter than it actually)

**Review by mstratford — COMMENTED** [review]
Will try and actually test the poetry bit, but otherwise looks ok.

File: README.md (hunk 1 above)
Comment (markspolakovs) [reply README.md:17]: Good catch!

Comment (markspolakovs) [reply README.md:17]:
```suggestion
- Python 3.7-3.10
```

**Comment by mstratford** [comment]
> It claims in the README that it's PEP8 compliant at least (and I think it's a bit stricter than it actually)

Curious, maybe that's recent, I seem to remember pep8 and black disagreeing, but no worries :)

**Comment by mstratford** [comment]
Mypy is giving me loads of stuff, haven't been able to confirm exactly what's going on.
Screenshot 2023-03-10 at 00 01 59

**Comment by mstratford** [comment]
> Mypy is giving me loads of stuff, haven't been able to confirm exactly what's going on. Screenshot 2023-03-10 at 00 01 59

Figured this out, poetry was using 3.11. Happy this is working.

**Review by mstratford — APPROVED** [review]

**Pull request merged**